### ISSUE TITLE
PairRDDFunctions vs esRDD & nested Maps

### ISSUE THREAD

**Issue by jeffsteinmetz** [elastic/elasticsearch-hadoop#284]
Is there documentation on how an esRDD compares to PairRDDFunctions?

Here is how I arrived at this question.
I am trying to pass a specific value from each ElasticSearch document to Hashing function (to later be used by Spark MLlib)

The maps returned from an esRDD contain a Map[String,Any]
Where String is the ID of the elasticsearch document and Any is another Map

PairRDDFunctions include methods that can operate in parallel, such as
`collectAsMap(): Map[K, V]`

For example, an Elasticsearch document returned from esRDD may look like:
`Map("ID1234567890" -> Map("title"-> "awesome links", "content" -> "This is the body of text from the author"))`

What we want is the value for the "title" or "content" key (to pass to a Hashing function)
`val newMap = esRDD.value`
returns a MapLike, but the second map is typed as ANY.  Which means this isn't a valid call:  `newMap("content")`

(not to mention .value probably returns everything to the master)
I am looking for the best design pattern to do things the spark way, while still being able to access the nested Map and iterate over potentially thousands of "content" strings.

In summary - trying to find a design pattern to best access values from all nested maps (which may even have other nested maps - i.e. complex JSON posted to ElasticSearch) from esRDD.  

I've had my head in scala/Play/Jackson JSON, so this is sounding similar to a JSONObject
( json \ "title" \ "subtitle"  )
